### PR TITLE
feat(wallet-api): Support EVM family in wallet-api

### PR DESCRIPTION
**Pull request by JunichiSugiura**
### 📝 Description

Convert Wallet API Ethereum transaction into Live Ethereum/EVM transaction based on Live account family.

### ❓ Context

- **Impacted projects**: `LLD`, `LLM`, `live-common`
- **Linked resource(s)**: https://github.com/LedgerHQ/ledger-live-issues/issues/108

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

> For trying out Polygon, you need to change its family from `ethereum` to `evm` on Ledger Live side [line](https://github.com/LedgerHQ/ledger-live/blob/d79a0b1b089ee2ff37181589ab898541188fb9fd/libs/ledgerjs/packages/cryptoassets/src/currencies.ts#L2241)

<details>
<summary>`manifest-local.json`</summary>

```
{
  "id": "paraswap-2",
  "name": "Paraswap",
  "private": false,
  "url": "http://localhost:3000/v2",
  "params": {
    "dappUrl": "https://embedded.paraswap.io/?referrer=ledger2&embed=true&enableStaking=false&displayMenu=false",
    "nanoApp": "Paraswap",
    "dappName": "Paraswap",
    "networks": [
      {
        "currency": "ethereum",
        "chainID": 1,
        "nodeURL": "https://eth-dapps.api.live.ledger.com"
      },
      {
        "currency": "bsc",
        "chainID": 56,
        "nodeURL": "https://bsc-dataseed.binance.org/"
      },
      {
        "currency": "polygon",
        "chainID": 137,
        "nodeURL": "https://polygon-mainnet.g.alchemy.com/v2/oPIxZM7kXsPVVY1Sk0kOQwkoIOpSu8PE"
      },
      {
        "currency": "fantom",
        "chainID": 250,
        "nodeURL": "https://fantom.publicnode.com"
      }
    ]
  },
  "homepageUrl": "https://paraswap.io",
  "supportUrl": "https://paraswap.io",
  "icon": "https://cdn.live.ledger.com/icons/platform/paraswap.png",
  "platform": "all",
  "apiVersion": "^2.0.0 || ~0.0.1",
  "manifestVersion": "1",
  "branch": "stable",
  "categories": ["nft"],
  "currencies": ["ethereum", "polygon", "fantom"],
  "content": {
    "shortDescription": {
      "en": "Swap your crypto with ParaSwap that aggregates and provides the best quotes decentralised exchanges."
    },
    "description": {
      "en": "Swap your crypto with ParaSwap that aggregates and provides the best quotes decentralised exchanges."
    }
  },
  "permissions": [
    "account.list",
    "account.receive",
    "account.request",
    "currency.list",
    "device.close",
    "device.exchange",
    "device.transport",
    "message.sign",
    "transaction.sign",
    "transaction.signAndBroadcast",
    "storage.set",
    "storage.get",
    "bitcoin.getXPub",
    "wallet.capabilities",
    "wallet.userId",
    "wallet.info"
  ],
  "domains": ["https://*"]
}
```
</details>

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: libs/ledger-live-common/src/wallet-api/helpers.ts [hunk 1]
```diff
@@ -13,7 +13,8 @@ export function isWalletAPISupportedCurrency(
   currency: Currency
 ): currency is WalletAPISupportedCurrency {
   if (isCryptoCurrency(currency)) {
-    return includes(WALLET_API_FAMILIES, currency.family);
+    // Todo: Paraswap UI won't render because `useCurrencies` returns empty array for some reasons. Need to investigate.
```
Comment (JunichiSugiura) [inline libs/ledger-live-common/src/wallet-api/helpers.ts:16]: 📌

File: libs/ledgerjs/packages/cryptoassets/src/currencies.ts [hunk 2]
```diff
@@ -3708,6 +3708,8 @@ export const cryptocurrenciesById: Record<CryptoCurrencyId, CryptoCurrency> = {
     ticker: "FTM",
     scheme: "fantom",
     color: "#1969ff",
+    // Todo: temporary set to test evm family
```
Comment (JunichiSugiura) [inline libs/ledgerjs/packages/cryptoassets/src/currencies.ts:3711]: 📌

**Review by JunichiSugiura — COMMENTED** [review]
Some lines to figure out next.

File: libs/ledger-live-common/src/families/ethereum/walletApiAdapter.ts [hunk 3]
```diff
@@ -1,44 +1,71 @@
...
+import { Transaction as EvmTx } from "../evm/types";
+import { isTokenAccount } from "../../account";
 
 const CAN_EDIT_FEES = true;
 
+type Transaction = EthTx | EvmTx;
+
 const areFeesProvided: AreFeesProvided<WalletAPIEthereumTransaction> = (tx) =>
-  !!(tx.gasLimit || tx.gasPrice);
+  !!(tx.gasLimit || tx.gasPrice || tx.maxFeePerGas || tx.maxPriorityFeePerGas);
```
Comment (IAmMorrow) [inline libs/ledger-live-common/src/families/ethereum/walletApiAdapter.ts:17]: Maybe we should be a little more strict with partial provided fee info here ? 

Either (tx.gasLimit && tx.gasPrice) || (tx.gasLimit && tx.maxFeePerGas && tx.maxPriorityFeePerGas) should be true ? That's an open question on if we want to support partial tx fee hints

Comment (JunichiSugiura) [reply libs/ledger-live-common/src/families/ethereum/walletApiAdapter.ts]: Good point, I applied.

File: libs/ledger-live-common/src/wallet-api/logic.ts [hunk 4]
```diff
@@ -111,15 +111,27 @@ export function signTransactionLogic(
...
-  if (accountFamily !== transaction.family) {
+  if (
+    accountFamily !== transaction.family &&
+    !(accountFamily === "evm" && transaction.family === "ethereum")
+  ) {
+    console.log(
+      "hey2",
+      { accountFamily, tx: transaction.family },
+      !(accountFamily === "evm" && transaction.family === "ethereum")
+    );
```
Comment (JunichiSugiura) [inline libs/ledger-live-common/src/wallet-api/logic.ts:122]: ✂️

**Review by ComradeAERGO — APPROVED** [review]
Lgtm

**Comment by JunichiSugiura** [comment]
ah, this PR is affected by this change: https://github.com/LedgerHQ/ledger-live/pull/3536

**Pull request merged**